logstash-core/lib/logstash/compiler/lscl/lscl_grammar.treetop

```diff
@@ -7,7 +7,7 @@ grammar LogStashCompilerLSCLGrammar
   end
 
   rule comment
-    (whitespace? "#" [^\r\n]* "\r"? "\n")+ <LogStash::Compiler::LSCL::AST::Comment>
+    (whitespace? "#" [^\r\n]* "\r"? ("\n" / !.))+ <LogStash::Compiler::LSCL::AST::Comment>
```

Comment:
Could you explain better why this change is necessary? 
```
("\n" / !.)
```
Should mean "newline character **or**  anything that is not a character".

Reply:
Do you think this requires a comment in the grammar?

Reply:
I thought was something along the EOF thing, so in this case you capture the comment at the end of file which terminates with EOF.

Would be better to put a comment on above the regexp to explain that `("\n" / !.)` is intended to capture the \n or the EOF in the comments that are the last in the file.

Reply:
changed this into a new rule and added a comment